### PR TITLE
Refactor with hooks

### PR DESCRIPTION
**Pull request by abrormukimov**
# In this milestone:

- I refactored class components with react hooks

### CODE REVIEW

File: src/components/Buttons/Input.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Emmyn5600) [inline src/components/Buttons/Input.js:1]: - [ ] Don't disable eslint errors, kindly fix them without disabling them.

File: src/components/Buttons/Numbers.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Emmyn5600) [inline src/components/Buttons/Numbers.js:1]: - [ ] Don't disable eslint errors, kindly fix them without disabling them.

File: src/components/Buttons/SimpleOperations.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Emmyn5600) [inline src/components/Buttons/SimpleOperations.js:1]: - [ ] Don't disable eslint errors, kindly fix them without disabling them.

File: src/components/Calculator.js [hunk 4]
```diff
@@ -1,35 +1,29 @@
...
-          <li className="span orange">+</li>
-          <li className="span2 gray">0</li>
-          <li className="span gray">.</li>
-          <li className="span orange">=</li>
-        </ul>
-      </div>
-    );
-  }
+function Calculator() {
+  const [currentInnerText, setInnerText] = useState('');
```
Comment (Emmyn5600) [inline src/components/Calculator.js:10]: - [ ] Let us use ES6 syntax for this function.

File: src/components/Buttons/SimpleOperations.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import PropTypes from 'prop-types';
+import classes from './SimpleOperations.module.css';
+
+function SimpleOperations(props) {
+  const { onclick } = props;
```
Comment (Emmyn5600) [inline src/components/Buttons/SimpleOperations.js:6]: - [ ] Let us use ES6 syntax for this function.

File: src/components/Buttons/Numbers.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import PropTypes from 'prop-types';
+import classes from './Numbers.module.css';
+
+function Numbers(props) {
+  const numbers = Array.from(Array(10).keys());
```
Comment (Emmyn5600) [inline src/components/Buttons/Numbers.js:6]: - [ ] Let us use ES6 syntax for this function.

File: src/components/Buttons/CalculatorOperations.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types';
+import classes from './CalculatorOperations.module.css';
+
+function CalculatorOperations(props) {
+  const { onclick } = props;
```
Comment (Emmyn5600) [inline src/components/Buttons/CalculatorOperations.js:6]: - [ ] Let us use ES6 syntax for this function.

File: src/components/Buttons/CalculatorOperations.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        <li><button type="button" onClick={onclick} name="+/-">+/-</button></li>
+        <li><button type="button" onClick={onclick} name="%">%</button></li>
+      </ul>
+    </div>
+  );
+}
+CalculatorOperations.propTypes = {
+  onclick: PropTypes.func.isRequired,
+};
+export default CalculatorOperations;
```
Comment (Emmyn5600) [inline src/components/Buttons/CalculatorOperations.js:20]: - [ ] Kindly, make sure that the sign next to the button name is coming after the button name not before. Refer to the way the calculator works as usual.

tdd1

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @abrormukimov 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️

- _Check the comment under the review._

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/components/Buttons/Input.js (hunk 1 above)
Comment (ZeenatLawal) [inline src/components/Buttons/Input.js:1]: - [ ] As suggested by the previous reviewer, kindly fix linter errors without disabling them. I would suggest running `npm i -S prop-types` to fix this linter issue.

File: src/components/Buttons/Numbers.js (hunk 2 above)
Comment (ZeenatLawal) [inline src/components/Buttons/Numbers.js:1]: - [ ] As suggested by the previous reviewer, kindly fix linter errors without disabling them. I would suggest running `npm i -S prop-types` to fix this linter issue.

File: src/components/Buttons/SimpleOperations.js (hunk 3 above)
Comment (ZeenatLawal) [inline src/components/Buttons/SimpleOperations.js:1]: - [ ] As suggested by the previous reviewer, kindly fix linter errors without disabling them. I would suggest running `npm i -S prop-types` to fix this linter issue.

File: src/components/Calculator.js [hunk 9]
```diff
@@ -1,35 +1,29 @@
...
-}
+const Calculator = () => {
+  const [currentInnerText, setInnerText] = useState('');
+  const [calc, setCalc] = useState({});
+  const changeHandler = (e) => {
+    const buttonName = e.target.name;
+    const newCalc = calculate(calc, buttonName);
+    setCalc(newCalc);
+    setInnerText(`${newCalc.next ? newCalc.next : ''} ${newCalc.operation ? newCalc.operation : ''} ${newCalc.total ? newCalc.total : ''}`);
+  };
```
Comment (ZeenatLawal) [inline src/components/Calculator.js:17]: - [ ] As suggested by the previous reviewer, kindly fix the arrangement of operations and numbers. I would suggest setting the inner text to `${newCalc.total ? newCalc.total : ''} ${newCalc.operation ? newCalc.operation : ''} ${newCalc.next ? newCalc.next : ''}` to resolve that.

File: src/components/Buttons/CalculatorOperations.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (ZeenatLawal) [inline src/components/Buttons/CalculatorOperations.js:1]: - [ ] As suggested by the previous reviewer, kindly fix linter errors without disabling them. I would suggest running `npm i -S prop-types` to fix this linter issue.

Comment (ZeenatLawal) [reply src/components/Buttons/CalculatorOperations.js:1]: - [OPTIONAL] It will be good to write a descriptive README following the Microverse template.

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]
Hi @abrormukimov, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ZeenatLawal — APPROVED** [review]
Hi @abrormukimov, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**